docs/source/glossary.md

```diff
@@ -0,0 +1,59 @@
+(glossary)=
+# PyTorch Glossary
+
+This glossary provides definitions for terms commonly used in PyTorch documentation.
+
+```{glossary}
+
+ATen
+   Short for "A Tensor Library". The foundational tensor and mathematical
+   operation library on which all else is built.
+
+Operation
+   A unit of work. For example, the work of matrix multiplication is an operation
+   called `aten::matmul`.
+
+Native Operation
+   An operation that comes natively with PyTorch ATen, for example `aten::matmul`.
+
+Custom Operation
+   An Operation that is defined by users and is usually a {term}`Compound Operation`.
+   For example, this [tutorial](https://pytorch.org/docs/stable/notes/extending.html)
+   details how to create Custom Operations.
+
+Kernel
+   Implementation of a PyTorch operation, specifying what should be done when an
+   operation executes.
```

Comment:
Refers to the "on device" implementation of a given operation. Reffering to the part of the Operation that actually computes the result.